### PR TITLE
[Comp19] Improve compile-time performance of RegisterCandidate

### PR DESCRIPTION
**Pull request by andrewcraik**
There are a number of inefficiencies in the implementation of functions in the OMR register candidate. This commit moves a number of small helper methods to inlines headers to allow inlining of small methods from extensible classes. It also cleans up some of the code in register candidate to reduce compile time.

This depends on Mem14-Comp18.

### CODE REVIEW

File: compiler/il/DataTypes.hpp [hunk 1]
```diff
@@ -24,7 +24,7 @@
 namespace TR
 {
```
Comment (vijaysun-omr) [inline compiler/il/DataTypes.hpp]: I would like @0xdaryl to review this commit as well.

File: compiler/il/OMRDataTypes.cpp [hunk 2]
```diff
@@ -56,92 +56,6 @@ static TR::ILOpCodes conversionMap[TR::NumOMRTypes][TR::NumOMRTypes] =
```
Comment (vijaysun-omr) [inline compiler/il/OMRDataTypes.cpp]: I would also like @0xdaryl to review this.

File: compiler/il/Symbol.hpp [hunk 3]
```diff
@@ -70,4 +70,6 @@ class OMR_EXTENSIBLE Symbol : public OMR::SymbolConnector
```
Comment (vijaysun-omr) [inline compiler/il/Symbol.hpp]: @0xdaryl could you please review ?

File: compiler/il/symbol/OMRSymbol.cpp [hunk 4]
```diff
@@ -145,895 +145,13 @@ OMR::Symbol::getRoundedSize()
    return roundedSize ? roundedSize : 4;
```
Comment (vijaysun-omr) [inline compiler/il/symbol/OMRSymbol.cpp]: @0xdaryl please review

File: tools/compiler/OMRRewriter/OMRRewriter.cpp [hunk 5]
```diff
@@ -342,7 +342,7 @@ class OMRThisRewritingVisitor : public RecursiveASTVisitor<OMRThisRewritingVisit
          trace("targetClass of static cast" << (targetClass ? targetClass->getQualifiedNameAsString() : "NULL" ) );
          CXXThisExpr *thisExpr = getThisExpr(cast->getSubExpr());
          if (thisExpr) {
-            CXXRecordDecl *thisConcrete = ClassChecker->getAssociatedConcreteType(thisExpr->getType()->getAs<PointerType>()->getPointeeType()->getAsCXXRecordDecl());
+            const CXXRecordDecl *thisConcrete = ClassChecker->getAssociatedConcreteType(thisExpr->getType()->getAs<PointerType>()->getPointeeType()->getAsCXXRecordDecl());
```
Comment (ncough) [inline tools/compiler/OMRRewriter/OMRRewriter.cpp]: This change does not seem relevant to the commit.

**Comment by ncough** [comment]
LGTM - Matches existing behaviour in RegisterCandidate changes and additional inlining should improve compile time.

**Comment by ncough** [comment]
This PR is missing changes to JitBuilder to pass the CI build. They are added in https://github.com/eclipse/omr/pull/1522.

**Comment by mstoodle** [comment]
This pull request includes a commit with an incomplete description of the change. On top of that, there is a breaking code change in this pull request. Unfortunately, a downstream project has an immediate deadline to include this change and @andrewcraik is unavailable to update this pull request.

To help out the downstream project and after discussion with @charliegracie, I will take the fairly drastic measure to summarily close this pull request to allow another contributor from that project will put forward an updated version with a fix for the breakage and fixing the commit message.

I do not want this to be a regular thing. In fact, now that it has happened, I want to state categorically that I do not want to be in a position requiring Eclipse OMR to ever do this again, certainly not for this particular project.
